lib/AST/DistributedDecl.cpp

```diff
@@ -76,6 +76,86 @@ bool swift::canSynthesizeDistributedActorCodableConformance(NominalTypeDecl *act
       false);
 }
 
+ExtensionDecl *
+swift::findDistributedActorAsActorExtension(
+    ProtocolDecl *distributedActorProto, ModuleDecl *module) {
+  ASTContext &C = distributedActorProto->getASTContext();
+  auto name = C.getIdentifier("__actorUnownedExecutor");
+  auto results = distributedActorProto->lookupDirect(
+      name, SourceLoc(),
+      NominalTypeDecl::LookupDirectFlags::IncludeAttrImplements);
+  for (auto result : results) {
+    if (auto var = dyn_cast<VarDecl>(result)) {
+      return dyn_cast<ExtensionDecl>(var->getDeclContext());
+    }
+  }
+
+  return nullptr;
+}
+
+bool swift::isDistributedActorAsLocalActorComputedProperty(VarDecl *var) {
+  auto &C = var->getASTContext();
+  return var->getName() == C.Id_asLocalActor &&
+         var->getDeclContext()->getSelfProtocolDecl() &&
+         var->getDeclContext()->getSelfProtocolDecl()->isSpecificProtocol(
+             KnownProtocolKind::DistributedActor);
+}
+
+VarDecl *
+swift::getDistributedActorAsLocalActorComputedProperty(ModuleDecl *module) {
+  auto &C = module->getASTContext();
+  auto DA = C.getDistributedActorDecl();
+  auto extension = findDistributedActorAsActorExtension(DA, module);
+
+  for (auto decl : extension->getMembers()) {
+    if (auto var = dyn_cast<VarDecl>(decl)) {
+      if (isDistributedActorAsLocalActorComputedProperty(var)) {
+        return var;
+      }
+    }
+  }
+
+  return nullptr;
+}
+
+ProtocolConformanceRef
+swift::getDistributedActorAsActorConformanceRef(ASTContext &C) {
+  auto actorProto = C.getProtocol(KnownProtocolKind::Actor);
+  auto distributedActorProto = C.getProtocol(KnownProtocolKind::DistributedActor);
+  Type distributedActorType = C.getDistributedActorType();
+
+  auto distributedActorAsActorConformance =
+      getDistributedActorAsActorConformance(C);
+
+  auto module = C.getStdlibModule();
+  VarDecl *asLocalActor =
+      getDistributedActorAsLocalActorComputedProperty(module);
+
+  ArrayRef<ProtocolConformanceRef> conformances = {
+    ProtocolConformanceRef(distributedActorProto)
+  };
+
+  auto subs =
+    SubstitutionMap::get(
+      asLocalActor->getDeclContext()->getGenericSignatureOfContext(),
+      MapTypeOutOfContext(),
+      LookUpConformanceInModule(actorProto->getParentModule()));
+
+    return ProtocolConformanceRef(
+        actorProto,
+        C.getSpecializedConformance(distributedActorType,
```

Comment:
Does this do anything? What happens if you add

distributedActorAsActorConformance->dump();
C.getSpecializedConformance(…)->dump();

Aren’t both the same?

Reply:
Thanks for the quick review and note, sorry I got caught up something urgent -- I'll check this as soon as i can

Reply:
So I checked and yes they're different, like so:

```
[DistributedDecl.cpp:144](getDistributedActorAsActorConformanceRef) THE distributedActorAsActorConformance ============
(normal_conformance type="\xCF\x84_0_0" protocol="Actor"
  (assoc_conformance type="Self" proto="AnyActor"
    (abstract_conformance protocol="AnyActor")))
```

```
[DistributedDecl.cpp:146](getDistributedActorAsActorConformanceRef) THE C.getSpecializedConformance(distributedActorType ============
(specialized_conformance type="DistributedActor" protocol="Actor"
  (substitution_map generic_signature=<Self where Self : DistributedActor>
    (substitution Self ->
      (generic_type_param_type depth=0 index=0 decl="Distributed.(file).DistributedActor.Self"))
    (conformance type="Self"
      (abstract_conformance protocol="DistributedActor")))
  (<conditional requirements unable to be computed>)
  (normal_conformance type="\xCF\x84_0_0" protocol="Actor"
    (assoc_conformance type="Self" proto="AnyActor"
      (abstract_conformance protocol="AnyActor"))))
[DistributedDecl.cpp:150](getDistributedActorAsActorConformanceRef)
```

Reply:
But it seems the project compiles fine if we just return `distributedActorAsActorConformance` directly huh -- should I prefer that then @slavapestov ? I don't know why a specialized_conformance or a normal_conformance might be preferred over another.

Reply:
I'll prepare the PR with just returning distributedActorAsActorConformance but would like to understand if for sure that's okey :)

Reply:
Yes because this is the identity substitution map:

```
  (substitution_map generic_signature=<Self where Self : DistributedActor>
    (substitution Self ->
      (generic_type_param_type depth=0 index=0 decl="Distributed.(file).DistributedActor.Self"))
    (conformance type="Self"
      (abstract_conformance protocol="DistributedActor")))
```

Reply:
Ah I see now, thank you!